File: backend/src/main/java/com/penocean/ehs/model/DailySafetyLog.java
package com.penocean.ehs.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DailySafetyLog {

    private Long id;
    private Long vesselId;
    private Long companyId;
    private LocalDate logDate;
    private String representativeName;
    private Integer attendeesCount;
    private String trainingContent;
    private String scannedFileUrl;
    private String kakaoMessageId;
    private String receivedVia;        // KAKAO / EMAIL / UPLOAD
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean deleted;
}

File: frontend/src/types/dailySafetyLog.ts
// Backend 계약: /api/daily-safety-logs
import type { PageResponse } from './common'

export type ReceivedVia = 'KAKAO' | 'EMAIL' | 'UPLOAD'

export interface DailySafetyLogResponse {
  id: number
  vesselId: number
  vesselName: string
  companyId: number
  companyName: string
  logDate: string
  representativeName: string
  attendeesCount: number
  trainingContent: string
  scannedFileUrl?: string
  receivedVia: ReceivedVia
  createdAt: string
}

export interface DailySafetyLogListParams {
  vesselId?: number
  dateFrom?: string
  dateTo?: string
  page?: number
  size?: number
}

export interface CreateDailySafetyLogPayload {
  vesselId: number
  companyId?: number
  logDate: string
  representativeName: string
  attendeesCount: number
  trainingContent: string
  receivedVia: ReceivedVia
}

export type DailySafetyLogPage = PageResponse<DailySafetyLogResponse>
